pvmismatch/pvmismatch_lib/pvmodule.py

```diff
@@ -484,18 +498,48 @@ def calcMod(self):
                     Isub, Vsub = self.pvconst.calcParallel(
                         Iparallel, Vparallel, Vparallel.max(), Vparallel.min()
                     )
-            bypassed = Vsub < self.Vbypass
-            Vsub[bypassed] = self.Vbypass
+                    
+            # apply bypass diodes depending on the configuration passed
+            if type(self.Vbypass) == list:
+                # if only one value is passed in the list- assume only one 
+                # bypass diode  across the PV module
+                if len(self.Vbypass) == 1: 
+                    bypass_entire_module = True
+                    Vbypass_mod = self.Vbypass[0]
+                # if more than 1 values are passed, apply them across  
+                # the cell strings in ascending order
+                elif len(self.cell_pos) == len(self.Vbypass):    
+                    if self.Vbypass[substr_idx] is None:
+                        bypassed = None
+                    else:
+                        bypassed = Vsub < self.Vbypass[substr_idx]
+                        Vsub[bypassed] = self.Vbypass[substr_idx]
+                else:
+                    raise Exception("wrong number of bypass diode values passed : %d"%(len(self.Vbypass)))
```

Comment:
seems like the wrong time to raise an exception, shouldn't this have happened when the PV module was instantiated? Why make the user wait to find out they screwed up, after they've already set everything up? IMO this will frustrate users and make them mad.